lib/vsc/install/shared_setup.py

```diff
@@ -664,43 +724,37 @@ def setup_sys_path(self):
         if os.path.isdir(REPO_SCRIPTS_DIR):
             sys.path.insert(0, REPO_SCRIPTS_DIR)
 
-        # insert lib dir
+        # insert lib dir before newly inserted test/base/scripts
         sys.path.insert(0, REPO_LIB_DIR)
 
-        # force __path__ of packages in the repo
-        packages = files_in_packages()['packages']
-        for package in packages.keys():
+        # force __path__ of packages in the repo (to deal with namespace extensions)
+
+        packages = FILES_IN_PACKAGES['packages']
+        # sort them, parents first
+        pkg_names = sorted(packages.keys())
+        # cleanup children first
+        reloaded_modules = []
+        for package in pkg_names[::-1]:
+            reloaded_modules.extend(self.reload_modules(package, remove_only=True, own_modules=True))
+
+        # insert in order, parents first
+        for package in pkg_names:
             try:
                 __import__(package)
+                log.debug('Imported package %s' % package)
             except ImportError as e:
                 raise ImportError("Failed to import package %s from current repository: %s" % (package, e))
             sys.modules[package].__path__.insert(0, os.path.dirname(packages[package][0]))
 
-        return cleanup
-
-    def reload_modules(self, package):
-        """
-        Cleanup and restore package because we use
-        vsc package tools very early.
-        So we need to make sure they are picked up from the paths as specified
-        in setup_sys_path, not to mix with installed and already loaded modules
-        """
-
-        def candidate(modulename):
-            """Select candidate modules to reload"""
-            return modulename in (package,) or modulename.startswith(package+'.')
-
-        loaded_modules = filter(candidate, sys.modules.keys())
-        reload_modules = []
-        for name in loaded_modules:
-            if hasattr(sys.modules[name], '__file__'):
-                # only actual modules
-                reload_modules.append(name)
-            del(sys.modules[name])
+        # reload the loaded modules with new __path__
+        for module in reloaded_modules:
+            try:
+                __import__(module)
+                log.debug('Imported module %s' % module)
+            except ImportError as e:
+                raise ImportError("Failed to reload module %s: %s" % (module, e))
```

Comment:
Given how `__import__('does_not_exist')` will raise `ImportError(`No module named %s' e)` do we really need to try, except and reraise with a new message? At this point, I think we can elide the noise and just call import and let the underlying machinery handle the error.